logrotate.c

```diff
@@ -582,8 +582,12 @@ static int createOutputFile(const char *fileName, int flags, const struct stat *
         return -1;
     }
 
-    if ((sb_create.st_uid != sb->st_uid || sb_create.st_gid != sb->st_gid) &&
-            fchown(fd, sb->st_uid, sb->st_gid)) {
+    /* Only attempt to set user/group if running as root */
+    if (
+        0 == geteuid() &&
```

Comment:
So this happens if an unprivileged user rotates a file of another unprivileged user?
Should this be supported by default, as the rotated log files are now having different ownership than the original log file?

Reply:
I see no reason why an unprivileged user should be prevented from rotating files they have access to.  Erroring out in this case could hardly help anybody.

Reply:
Indeed @kdudka.  The directory where I `logrotate` the container logs looks like:
    -rw-r--r-- 1 _apt root       0 Jan 28 01:02 error.log
    -rw-r--r-- 1 _apt root    7912 Jan 27 11:08 error.log.1
    -rw-r--r-- 1 ram  ram       20 Jan 26 01:02 error.log.2.gz
So yes, the log files are given to me, but then I have read-access to the logs and I could copy/rename/compress them myself, since I have write access to the directory containing them...  Anything I can do manually should be possible with logrorate...

Reply:
Convinced.
For consistency the check should be against `ROOT_UID` instead of plain `0`.